lib/datadog/grok/src/lexer.rs

```diff
@@ -258,19 +258,31 @@ fn is_digit(ch: char) -> bool {
 fn unescape_string_literal(mut s: &str) -> String {
     let mut string = String::with_capacity(s.len());
     while let Some(i) = s.bytes().position(|b| b == b'\\') {
-        let c = match s.as_bytes()[i + 1] {
-            b'\'' => '\'',
-            b'"' => '"',
-            b'\\' => '\\',
-            b'n' => '\n',
-            b'r' => '\r',
-            b't' => '\t',
-            _ => unimplemented!("invalid escape"),
-        };
-
-        string.push_str(&s[..i]);
-        string.push(c);
-        s = &s[i + 2..];
+        if s.bytes().len() > i + 2 {
```

Comment:
You should be able to just do `s.len()` here. 

https://doc.rust-lang.org/std/primitive.str.html#method.len

Reply:
Right, thanks, udpated.